packages/react-charting/src/components/CommonComponents/ErrorBoundary.base.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react';
+import { classNamesFunction, getId } from '@fluentui/react/lib/Utilities';
+import { IErrorBoundaryProps, IErrorBoundaryStyles } from './ErrorBoundary.types';
+import { ErrorBadgeIcon, ErrorIcon, StatusErrorFullIcon } from '@fluentui/react-icons-mdl2';
+
+export enum ErrorCodes {
+  NoData = 'No data',
+  GeneralError = 'General error',
+}
+
+interface IErrorBoundaryState {
+  hasError: boolean;
+  errorMsg: string;
+}
+const getClassNames = classNamesFunction<IErrorBoundaryProps, IErrorBoundaryStyles>();
+
+export class ErrorBoundaryBase extends React.Component<IErrorBoundaryProps, IErrorBoundaryState> {
+  private _defaultErrorMsg = "Something went wrong and we couldn't get the page to display";
+  private _classNames;
+  public static getDerivedStateFromError(error: Error) {
+    return { hasError: true, errorMsg: error.message };
+  }
+
+  constructor(props: IErrorBoundaryProps) {
+    super(props);
+    this.state = {
+      hasError: false,
+      errorMsg: '',
+    };
+    this._classNames = getClassNames(this.props.styles!, {
+      hasEmptyState: this.props.hasEmptyState,
+      theme: this.props.theme!,
+      width: this.props.width,
+      height: this.props.height,
+    });
+  }
+
+  public componentDidCatch(_error: any, errorInfo: any) {
+    this.setState({
+      hasError: true,
+      errorMsg: errorInfo,
+    });
+  }
+
+  public render() {
+    if (this.state.hasError || this.props.hasEmptyState) {
+      if (this.state.hasError && this.props.handleError !== undefined) {
+        return this.props.handleError();
+      }
+
+      if (this.props.hasEmptyState && this.props.handleEmptyState !== undefined) {
+        return (
+          <div
+            id={getId('_Chart_empty_')}
+            role={'alert'}
+            aria-label={this.state.errorMsg ? this.state.errorMsg : this._defaultErrorMsg}
+            className={this._classNames.root}
+          >
+            {this.props.handleEmptyState()}
+          </div>
+        );
+      }
+      return (
+        <div id={getId('_Chart_empty_')} className={this._classNames.root} aria-label={this._getAriaLabel()}>
```

Comment:
this id applies to error state also

Reply:
yes, kept the ids same because in both cases the chart will be empty